lib/matplotlib/testing/__init__.pyi

```diff
@@ -0,0 +1,50 @@
+from collections.abc import Callable
+import subprocess
+from typing import Any, IO, Literal, overload
+
+def set_font_settings_for_testing() -> None: ...
+def set_reproducibility_for_testing() -> None: ...
+def setup() -> None: ...
+@overload
+def subprocess_run_for_testing(
+    command: list[str],
+    env: dict[str, str] | None = ...,
+    timeout: float | None = ...,
+    stdout: int | IO[Any] | None = ...,
+    stderr: int | IO[Any] | None = ...,
+    check: bool = ...,
+    *,
+    text: Literal[True],
+    capture_output: bool = ...,
+) -> subprocess.CompletedProcess[str]: ...
+@overload
+def subprocess_run_for_testing(
+    command: list[str],
+    env: dict[str, str] | None = ...,
+    timeout: float | None = ...,
+    stdout: int | IO[Any] | None = ...,
+    stderr: int | IO[Any] | None = ...,
+    check: bool = ...,
+    *,
+    text: Literal[False],
```

Comment:
```suggestion
    text: Literal[False] = ...,
```

I think it's actually fine to leave this as with the default (and kw-or-positional) which will disambiguate the output in the case where text is not passed (which defaults to False)

Technically the same game with `bool` could be played as with `text=True` where you require it to be passed explicitly for the ambiguous case to enter into it, but don't think that is even needed... as it will match to the first matching overloaded signature, which would be the Literal[False] one unless it is passed.

Reply:
Yep, that seems to work as well.